### PR TITLE
Create component for handling questions of type 'BOOLEAN'

### PR DESCRIPTION
**Pull request by egracer**
Works on #15.

- Stores & pulls the user's response to the question in Redux.
- Takes a `showError` boolean as a prop, allowing the parent component to control error state during form validation.

### CODE REVIEW

File: client/src/components/BooleanQuestion.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const handleChange = (_, {value}) => {
+    dispatch(updateResponses(
+        {
+          questionKey: questionKey,
+          response: value,
+        }));
+  };
+
+  return (
+    <Form size='large'>
```
Comment (kebrahim) [inline client/src/components/BooleanQuestion.jsx:46]: do we need this Form now that michelle's PR has added the Form at the top level? https://github.com/ny/maslow/pull/53/files

File: client/src/components/BooleanQuestion.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            checked={responseData === 'true'} onChange={handleChange}
+          />
+          <Form.Radio
+            // TODO: translation
+            label='No' value='false'
+            className={radioItemClass} name={radioGroupName}
+            checked={responseData === 'false'} onChange={handleChange}
+          />
+          {showError &&
+            <Label pointing prompt>Please select yes or no.</Label>}
```
Comment (kebrahim) [inline client/src/components/BooleanQuestion.jsx:66]: TODO translation, right?

File: client/src/css/questions.css [hunk 3]
```diff
@@ -1,3 +1,9 @@
 .questionHint {
   font-size: 1rem;
 }
+
+.ui.form .grouped.fields .booleanQuestionRadioButton {
+  border: 1px solid lightgray;
+  padding: 10px;
+  margin-top: 10px;
+}
```
Comment (kebrahim) [inline client/src/css/questions.css:9]: add newline

File: client/src/components/BooleanQuestion.jsx (hunk 1 above)
Comment (egracer) [reply client/src/components/BooleanQuestion.jsx:46]: We do not! Removing.

File: client/src/components/BooleanQuestion.jsx (hunk 2 above)
Comment (egracer) [reply client/src/components/BooleanQuestion.jsx:66]: yes I'm actually doing this work now.

File: client/src/css/questions.css (hunk 3 above)
Comment (egracer) [reply client/src/css/questions.css:9]: done

File: client/src/components/BooleanQuestion.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        {
+          questionKey: questionKey,
+          response: value,
+        }));
+  };
+
+  return (
+    <Form size='large'>
+      <Form.Field required error={showError}>
+        <label>{question.text}</label>
```
Comment (marcja) [inline client/src/components/BooleanQuestion.jsx:48]: TODO translation

File: client/src/components/BooleanQuestion.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        {!!question.hint &&
+          <p className='questionHint'>{question.hint}</p>}
```
Comment (marcja) [inline client/src/components/BooleanQuestion.jsx:50]: TODO translation

File: client/src/components/BooleanQuestion.jsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  // TODO: replace with i18next
+  const question =
+    questionData ?
+      questionData.resources.en :
+      {text: '', hint: ''};
+
+  const handleChange = (_, {value}) => {
+    dispatch(updateResponses(
+        {
+          questionKey: questionKey,
```
Comment (marcja) [inline client/src/components/BooleanQuestion.jsx:40]: Like NumberQuestion, TODO by changing to {HOUSEHOLD_SIZE: 5}

Comment (egracer) [reply client/src/components/BooleanQuestion.jsx:40]: Done

File: client/src/components/BooleanQuestion.jsx (hunk 4 above)
Comment (egracer) [reply client/src/components/BooleanQuestion.jsx:48]: Done

File: client/src/components/BooleanQuestion.jsx (hunk 5 above)
Comment (egracer) [reply client/src/components/BooleanQuestion.jsx:50]: Done

**Pull request merged**